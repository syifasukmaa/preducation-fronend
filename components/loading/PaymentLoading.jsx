import React from 'react'

export default function PaymentLoading() {
  return (
    <tr className="my-10">
      <td className="px-4 w-[15%]">
        <p className="w-20 py-1 my-4 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="px-4 w-[15%]">
        <div className="flex flex-col justify-between h-full">
          <p className="py-1 rounded-sm animate-pulse bg-neutral-03"></p>
        </div>
      </td>
      <td className="px-4 w-[15%]">
        <p className="w-[60%] py-1 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="px-4 w-[15%]">
        <p className="py-1 rounded-sm w-[60%] animate-pulse bg-neutral-03"></p>
      </td>
      <td className="px-4 w-[20%]">
        <div className="flex flex-col justify-between h-full">
          <p className="py-1 mt-1 rounded-sm animate-pulse bg-neutral-03 w-[50%]"></p>
        </div>
      </td>
      <td className="px-4 w-[20%]">
        <div className="flex flex-col justify-between h-full">
          <p className="w-3/4 py-1 pl-4 mt-1 rounded-sm animate-pulse bg-neutral-03 lg:pl-0 lg:pr-1"></p>
        </div>
      </td>
    </tr>
  )
}
