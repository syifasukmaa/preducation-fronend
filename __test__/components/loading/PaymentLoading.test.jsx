import PaymentLoading from '@/components/loading/PaymentLoading'
import { render } from '@testing-library/react'

describe('Payment Loading', () => {
  it('renders loading skeleton', () => {
    const container = render(<PaymentLoading />)
    expect(container).toMatchSnapshot()
  })
})
