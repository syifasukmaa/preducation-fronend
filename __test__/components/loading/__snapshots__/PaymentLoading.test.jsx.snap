// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Loading renders loading skeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="my-10"
        data-testid="payment-loadingundefined"
      >
        <td
          class="px-4 w-[15%]"
        >
          <p
            class="w-20 py-1 my-4 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
          />
        </td>
        <td
          class="px-4 w-[15%]"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="py-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
            />
          </div>
        </td>
        <td
          class="px-4 w-[15%]"
        >
          <p
            class="w-[60%] py-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
          />
        </td>
        <td
          class="px-4 w-[15%]"
        >
          <p
            class="py-1 rounded-sm w-[60%] animate-pulse bg-neutral-03 dark:bg-neutral-04"
          />
        </td>
        <td
          class="px-4 w-[20%]"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="py-1 mt-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04 w-[50%]"
            />
          </div>
        </td>
        <td
          class="px-4 w-[20%]"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="w-3/5 py-1 pl-4 mt-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04 lg:pl-0 lg:pr-1"
            />
          </div>
        </td>
        <td
          class="pl-8 w-[10%]"
        >
          <p
            class="py-1 rounded-sm w-[100%] animate-pulse bg-neutral-03 dark:bg-neutral-04"
          />
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="my-10"
      data-testid="payment-loadingundefined"
    >
      <td
        class="px-4 w-[15%]"
      >
        <p
          class="w-20 py-1 my-4 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
        />
      </td>
      <td
        class="px-4 w-[15%]"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="py-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
          />
        </div>
      </td>
      <td
        class="px-4 w-[15%]"
      >
        <p
          class="w-[60%] py-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04"
        />
      </td>
      <td
        class="px-4 w-[15%]"
      >
        <p
          class="py-1 rounded-sm w-[60%] animate-pulse bg-neutral-03 dark:bg-neutral-04"
        />
      </td>
      <td
        class="px-4 w-[20%]"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="py-1 mt-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04 w-[50%]"
          />
        </div>
      </td>
      <td
        class="px-4 w-[20%]"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="w-3/5 py-1 pl-4 mt-1 rounded-sm animate-pulse bg-neutral-03 dark:bg-neutral-04 lg:pl-0 lg:pr-1"
          />
        </div>
      </td>
      <td
        class="pl-8 w-[10%]"
      >
        <p
          class="py-1 rounded-sm w-[100%] animate-pulse bg-neutral-03 dark:bg-neutral-04"
        />
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
